hapi-engine requires hapi instead of @hapi/hapi as peer dependency

# 🐞 Bug report

#### What modules are related to this issue?
<!-- ✍️edit: -->
- [ ] aspnetcore-engine
- [ ] builders
- [ ] common
- [ ] express-engine
- [x] hapi-engine

### Is this a regression?

<!-- Did this behavior used to work in the previous version? -->
Kind of. Since v7 npm will try to install peer dependencies and does not complete the installation if the peer dependencies can't be installed.

### Description

[`hapi`](https://www.npmjs.com/package/hapi) is deprecated and not maintained anymore. The package is now called [`@hapi/hapi`](https://www.npmjs.com/package/@hapi/hapi).

## 🔬 Minimal Reproduction

```sh
npx @angular/cli new example # this needs to be done with npm@6 because it fails as well with npm@7
cd example
ng generate universal --client-project example
npm i npm@7 --global
npm i @nguniversal/hapi-engine
```

## 🔥 Exception or Error
<pre><code>
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: hapi@undefined
npm WARN node_modules/hapi
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer hapi@"^18.4.0" from @nguniversal/hapi-engine@11.1.1
npm WARN node_modules/@nguniversal/hapi-engine
npm WARN   @nguniversal/hapi-engine@"*" from the root project
npm ERR! code ETARGET
npm ERR! notarget No matching version found for hapi@^18.4.0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
</code></pre>


## 🌍 Your Environment
<pre><code>
Angular CLI: 11.1.2
Node: 12.20.1
OS: darwin x64

Angular: 11.1.1
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1101.2
@angular-devkit/build-angular   0.1101.2
@angular-devkit/core            11.1.2
@angular-devkit/schematics      11.1.2
@angular/cli                    11.1.2
@schematics/angular             11.1.2
@schematics/update              0.1101.2
rxjs                            6.6.3
typescript                      4.1.3
</code></pre>

--- a/modules/hapi-engine/package.json
+++ b/modules/hapi-engine/package.json
@@ -15,7 +15,7 @@
     "@angular/common": "NG_VERSION",
     "@angular/core": "NG_VERSION",
     "@angular/platform-server": "NG_VERSION",
-    "hapi": "HAPI_VERSION"
+    "@hapi/hapi": "HAPI_VERSION"
   },
   "dependencies": {
     "@nguniversal/common": "0.0.0-PLACEHOLDER",

